### PR TITLE
Guard against division by zero for no VGPR usage (e.g., in an empty k…

### PR DESCRIPTION
**Pull request by arghdos**
…ernel)

### CODE REVIEW

File: src/hip_module.cpp [hunk 1]
```diff
@@ -1326,7 +1326,8 @@ hipError_t ihipOccupancyMaxActiveBlocksPerMultiprocessor(
     size_t numWavefronts = (blockSize + wavefrontSize - 1) / wavefrontSize;
 
     size_t availableVGPRs = (prop.regsPerBlock / wavefrontSize / simdPerCU);
-    size_t vgprs_alu_occupancy = simdPerCU * std::min(maxWavesPerSimd, availableVGPRs / usedVGPRS);
+    size_t vgprs_alu_occupancy = simdPerCU * ((usedVGPRS == 0) ? maxWavesPerSimd
```
Comment (emankov) [inline src/hip_module.cpp]: LGTM, otherwise excessive brackets.

Comment (arghdos) [reply src/hip_module.cpp]: I actually used the bracket format from the SGPR check a few lines down :p
Should I change both?

Comment (arghdos) [reply src/hip_module.cpp]: Actually, after removing I get gcc errors:

>/home/research/code/HIP/src/hip_module.cpp:1329:63: warning: operator '?:' has lower precedence than '*'; '*' will be evaluated first [-Wparentheses]
>    size_t vgprs_alu_occupancy = simdPerCU * (usedVGPRS == 0) ? maxWavesPerSimd

looks like we need to keep them

Comment (emankov) [reply src/hip_module.cpp]: There were other (inner) brackets in mind, actually, smth. like:
```c++
size_t vgprs_alu_occupancy = simdPerCU * (usedVGPRS ? std::min(maxWavesPerSimd, availableVGPRs / usedVGPRS) : maxWavesPerSimd);
```

Comment (arghdos) [reply src/hip_module.cpp]: gotcha

**Pull request merged**